<!-- 
The MIT License (MIT)
Copyright (c) 2016 GRVC University of Seville

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
-->
<launch>
    <arg name="save_data" default="false"/>
    <arg name="multi" default="false"/>
    <arg name="robot_model" default="mbzirc"/>
    <arg name="trajectory" default="false"/>
    <arg name="vxy" default="2.0"/>
    <arg name="vz_up" default="3.0"/>
    <arg name="vz_dn" default="1.0"/>
    <arg name="ns_prefix" default="uav_"/>

    <arg name="pub_rate" default="50.0"/>



    <!-- UAV Path Manager nodes -->
    <group ns="uav_path_manager">
        <group ns="$(arg ns_prefix)1">
            <node pkg="uav_path_manager" type="path_generator_node" name="generator" output="screen" required="true">
                <param name="vxy" value="$(arg vxy)"/>
                <param name="vz_up" value="$(arg vz_up)"/>
                <param name="vz_dn" value="$(arg vz_dn)"/>
                 <param name="pub_rate" value="$(arg pub_rate)"/>
            </node>
            <node pkg="uav_path_manager" type="path_manager_node" name="manager" output="screen" required="true">
                <param name="uav_id" value="1"/>
                <param name="save_csv" value="$(arg save_data)"/>
                <param name="trajectory" value="$(arg trajectory)"/>
                 <param name="pub_rate" value="$(arg pub_rate)"/>
                <param name="csv_file" value="wo_obst"/>

            </node>
            <node pkg="uav_path_manager" type="path_follower_node" name="follower" output="screen" required="true">
                <param name="uav_id" value="1"/>
                 <param name="pub_rate" value="$(arg pub_rate)"/>
            </node>
        </group>

        <!-- Add one more UAV and nodes to manage it -->
        <group if="$(arg multi)" ns="$(arg ns_prefix)2">
            <node pkg="uav_path_manager" type="path_manager_node" name="manager" required="true" output="screen">
                <param name="uav_id" value="2"/>
                <param name="trajectory" value="$(arg trajectory)"/>
                 <param name="pub_rate" value="$(arg pub_rate)"/>
            </node>
            <node pkg="uav_path_manager" type="path_follower_node" name="follower" required="true">
                <param name="uav_id" value="2"/>
                 <param name="pub_rate" value="$(arg pub_rate)"/>
            </node>
        </group>
    </group>

</launch>
